### ISSUE TITLE
undefined method `breadcrumb_parent' for nil:NilClass

### ISSUE THREAD

**Issue by thelastinuit** [sferik/rails_admin#1887]
I'm using the rails_admin_import addon but the error i got it's from rails_admin.
I saw the previous issue with the same name but didn't help.
Anyone?

this is the config:
# -_\- encoding : utf-8 -_-
# -_\- coding: utf-8 -_-

require 'i18n'
I18n.default_locale = :es
# RailsAdmin config file. Generated on August 22, 2013 15:42
# See github.com/sferik/rails_admin for more informations

RailsAdmin.config do |config|

  config.actions do
    all
    charts
    import
  end
  ################  Global configuration  ################
  # Set the admin name here (optional second array element will appear in red). For example:
  config.main_app_name = ['Ternium Access |', 'Dashboard']
  # or for a more dynamic name:
  # config.main_app_name = Proc.new { |controller| [Rails.application.engine_name.titleize, controller.params['action'].titleize] }

  # RailsAdmin may need a way to know who the current user is]
  config.current_user_method { current_user } # auto-generated

  # If you want to track changes on your models:
  #config.audit_with :paper_trail, 'User'
  #config.audit_with :paper_trail, 'HistoryEvent'
  # Or with a PaperTrail: (you need to install it first)
  # config.audit_with :paper_trail, 'User'

  # Display empty fields in show views:
  config.compact_show_view = false

  # Number of default rows per-page:
  config.default_items_per_page = 20

  # CanCan
  config.authorize_with :cancan

  # Exclude specific models (keep the others):
  # config.excluded_models = ['Role', 'User']

  # Include specific models (exclude the others):
  # config.included_models = ['Role', 'User']

  # Label methods for model instances:
  # config.label_methods << :description # Default is [:name, :title]

  ################  Model configuration  ################

  # Each model configuration can alternatively:
  #   - stay here in a `config.model 'ModelName' do ... end` block
  #   - go in the model definition file in a `rails_admin do ... end` block

  # This is your choice to make:
  #   - This initializer is loaded once at startup (modifications will show up when restarting the application) but all RailsAdmin configuration would stay in one place.
  #   - Models are reloaded at each request in development mode (when modified), which may smooth your RailsAdmin development workflow.

  # Now you probably need to tour the wiki a bit: https://github.com/sferik/rails_admin/wiki
  # Anyway, here is how RailsAdmin saw your application's models when you ran the initializer:

  ###  Role  ###

  # config.model 'Role' do

  #   # You can copy this to a 'rails_admin do ... end' block inside your role.rb model definition

  #   # Found associations:

  #     configure :resource, :polymorphic_association         # Hidden 
  #     configure :users, :has_and_belongs_to_many_association 

  #   # Found columns:

  #     configure :id, :integer 
  #     configure :name, :string 
  #     configure :resource_id, :integer         # Hidden 
  #     configure :resource_type, :string         # Hidden 
  #     configure :created_at, :datetime 
  #     configure :updated_at, :datetime 

  #   # Cross-section configuration:

  #     # object_label_method :name     # Name of the method called for pretty printing an _instance_ of ModelName
  #     # label 'My model'              # Name of ModelName (smartly defaults to ActiveRecord's I18n API)
  #     # label_plural 'My models'      # Same, plural
  #     # weight 0                      # Navigation priority. Bigger is higher.
  #     # parent OtherModel             # Set parent model for navigation. MyModel will be nested below. OtherModel will be on first position of the dropdown
  #     # navigation_label              # Sets dropdown entry's name in navigation. Only for parents!

  #   # Section specific configuration:

  #     list do
  #       # filters [:id, :name]  # Array of field names which filters should be shown by default in the table header
  #       # items_per_page 100    # Override default_items_per_page
  #       # sort_by :id           # Sort column (default is primary key)
  #       # sort_reverse true     # Sort direction (default is true for primary key, last created first)
  #     end
  #     show do; end
  #     edit do; end
  #     export do; end
  #     # also see the create, update, modal and nested sections, which override edit in specific cases (resp. when creating, updating, modifying from another model in a popup modal or modifying from another model nested form)
  #     # you can override a cross-section field configuration in any section with the same syntax `configure :field_name do ... end`
  #     # using `field` instead of `configure` will exclude all other fields and force the ordering
  # end

  ###  User  ###

  # config.model 'User' do

  #   # You can copy this to a 'rails_admin do ... end' block inside your user.rb model definition

  #   # Found associations:

  #     configure :roles, :has_and_belongs_to_many_association 

  #   # Found columns:

  #     configure :id, :integer 
  #     configure :email, :string 
  #     configure :password, :password         # Hidden 
  #     configure :password_confirmation, :password         # Hidden 
  #     configure :reset_password_token, :string         # Hidden 
  #     configure :reset_password_sent_at, :datetime 
  #     configure :remember_created_at, :datetime 
  #     configure :sign_in_count, :integer 
  #     configure :current_sign_in_at, :datetime 
  #     configure :last_sign_in_at, :datetime 
  #     configure :current_sign_in_ip, :string 
  #     configure :last_sign_in_ip, :string 
  #     configure :created_at, :datetime 
  #     configure :updated_at, :datetime 
  #     configure :name, :string 

  #   # Cross-section configuration:

  #     # object_label_method :name     # Name of the method called for pretty printing an _instance_ of ModelName
  #     # label 'My model'              # Name of ModelName (smartly defaults to ActiveRecord's I18n API)
  #     # label_plural 'My models'      # Same, plural
  #     # weight 0                      # Navigation priority. Bigger is higher.
  #     # parent OtherModel             # Set parent model for navigation. MyModel will be nested below. OtherModel will be on first position of the dropdown
  #     # navigation_label              # Sets dropdown entry's name in navigation. Only for parents!

  #   # Section specific configuration:

  #     list do
  #       # filters [:id, :name]  # Array of field names which filters should be shown by default in the table header
  #       # items_per_page 100    # Override default_items_per_page
  #       # sort_by :id           # Sort column (default is primary key)
  #       # sort_reverse true     # Sort direction (default is true for primary key, last created first)
  #     end
  #     show do; end
  #     edit do; end
  #     export do; end
  #     # also see the create, update, modal and nested sections, which override edit in specific cases (resp. when creating, updating, modifying from another model in a popup modal or modifying from another model nested form)
  #     # you can override a cross-section field configuration in any section with the same syntax `configure :field_name do ... end`
  #     # using `field` instead of `configure` will exclude all other fields and force the ordering
  # end
  # Date error fix

  config.included_models = ["User", "Event", "Venue", "HistoryEvent", "Role", "Transport"]

  #config.model Ckeditor::Assets do
  #  list do 
  #    field :data_file_name do
  #      label "Nombre de archivo"
  #    end
  #    field :data_content_type do
  #      label "Tipo"
  #    end
  #    field :data_file_size do
  #      label "Tamano"
  #    end
  #    field :width do
  #      label "Ancho"
  #    end
  #    field :height do
  #      label "Alto"
  #    end
  #  end
  #  show do 
  #    field :data_file_name do
  #      label "Nombre de archivo"
  #    end
  #    field :data_content_type do
  #      label "Tipo"
  #    end
  #    field :data_file_size do
  #      label "Tamano"
  #    end
  #    field :width do
  #      label "Ancho"
  #    end
  #    field :height do
  #      label "Alto"
  #    end  
  #    field :created_at do
  #      label "Creado"
  #    end
  #  end
  #end
  #config.model Ckeditor::AttachmentFile do
  #  label "Anexo" 
  #  label_plural "Anexos"
  #  list do 
  #    field :data do
  #      label "Tipo"
  #    end
  #    field :width do
  #      label "Ancho"
  #    end
  #    field :height do
  #      label "Alto"
  #    end
  #  end
  #  show do 
  #    field :data do
  #      label "Tipo"
  #    end
  #    field :width do
  #      label "Ancho"
  #    end
  #    field :height do
  #      label "Alto"
  #    end
  #    field :created_at do
  #      label "Creado"
  #    end
  #  end
  #end

  #config.model Ckeditor::Picture do
  #  label "Imagen" 
  #  label_plural "Imagenes"
  #  list do 
  #    field :data do
  #      label "Tipo"
  #    end
  #    field :width do
  #      label "Ancho"
  #    end
  #    field :height do
  #      label "Alto"
  #    end
  #  end
  #  show do 
  #    field :data do
  #      label "Tipo"
  #    end
  #    field :width do
  #      label "Ancho"
  #    end
  #    field :height do
  #      label "Alto"
  #    end
  #    field :created_at do
  #      label "Creado"
  #    end
  #  end
  #end
  config.model 'Transport' do
    label "Transporte" 
    label_plural "Transportes"
    list do
      field :name do
        label "Transporte"
      end
      field :event do
        label "Evento"
      end
    end
    show do
      field :name do
        label "Transporte"
      end
      field :event do
        label "Evento"
      end
    end
    edit do
      field :name do
        label "Transporte"
      end
      field :event do
        label "Evento"
      end
    end
  end

  config.model 'Transfer' do
    rails_admin do
       label "Transferencia" 
       label_plural "Transferencias"
       list do
         field :name do
           label "Transferencia"
         end
         field :event do
           label "Evento"
         end
       end
       show do
         field :name do
           label "Transferencia"
         end
         field :event do
           label "Evento"
         end
       end
       edit do
         field :name do
           label "Transferencia"
         end
         field :event do
           label "Evento"
         end
       end
     end
  end
  config.model 'User' do
        label "Usuario" 
        label_plural "Usuarios"
        list do
          #field :id do
          #  label "Folio"
          #end
          field :name do
            label "Nombre"
          end
          field :lastname do
            label "Apellidos"
          end
          field :company do
            label "Compañía"
          end
          field :address do
            label "Direccion"
          end
          field :email do
            label "E-mail"
          end
          field :telephone do
            label "Teléfono"
          end
          field :mobile do
            label "Móvil"
          end
          field :professional_information do
            label "Información Adicional"
          end
          field :roles do
              label "Rol"
            end
      #    field :events do
          #  label "Eventos"
          #  inverse_of :users
          #end
        #   field :venues do
        #    label "Sedes"
        #    inverse_of :users
        #  end
          field :barcode do
            label "Identificador"
          end
        end
        show do
          field :id do
            label "Folio"
          end
          field :name do
            label "Nombre"
          end
          field :lastname do
            label "Apellidos"
          end
          field :company do
            label "Compania"
          end
          field :address do
            label "Direccion"
          end
          field :email do
            label "E-mail"
          end
          field :telephone do
            label "Teléfono"
          end
          field :mobile do
            label "Móvil"
          end
          field :professional_information do
            label "Información Adicional"
          end
          field :events do
            label "Eventos"
            inverse_of :users
          end
          field :venues do
            label "Sedes"
            inverse_of :users
          end
          field :roles do
            label "Rol"
          end
          #field :barcode do
          #  label "QR Code"
          #  pretty_value do # used in list view columns and show views, defaults to formatted_value for non-association fields
          #    bindings[:view].tag(:img, { :src => '/assets/' + value.to_s + '.png' })
          #  end
          #end
        end
        edit do
          field :name do
            label "Nombre"
          end
          field :lastname do
            label "Apellidos"
          end
          field :company do
            label "Compañía"
          end
          field :address do
            label "Dirección"
          end
          field :email do
            label "E-mail"
          end
          field :password do
            label "Contraseña"
          end
          field :password_confirmation do
            label "Repetir contrasena"
          end
          field :roles do
            label "Rol"
          end
          field :telephone do
            label "Teléfono"
          end
          field :mobile do
            label "Móvil"
          end
          field :events do
            label "Eventos"
            inverse_of :users
          end
          field :venues do
            label "Sedes"
            inverse_of :users
          end
          field :professional_information do
            label "Información Adicional"
          end  
        end
  end

  config.model 'Venue' do
    label "Sede" 
    label_plural "Sedes"
    list do
      #field :id do
      #  label "Sede ID"
      #end
      field :name do
        label "Nombre"
      end
      field :address do
        label "Direccion"
      end
      field :telephone do
        label "Telefono"
      end
    end
    show do
      field :id do
        label "Sede ID"
      end
      field :name do
        label "Nombre"
      end
      field :address do
        label "Direccion"
      end
      field :telephone do
        label "Telefono"
      end
    end
  end

  config.model 'Event' do
    label "Evento" 
    label_plural "Eventos"
    list do
      #field :id do
      #  label "Evento ID"
      #end
      field :name do
        label "Nombre"
      end
      field :start_date do
        label "Fecha de Inicio"
        date_format :default
      end
      field :end_date do
        label "Fecha de Cierre"
        date_format :default
      end
    end
    show do
      field :id do
        label "Evento ID"
      end
      field :name do
        label "Nombre"
      end
      field :start_date do
        label "Fecha de Inicio"
        date_format :default
      end
      field :end_date do
        label "Fecha de Cierre"
        date_format :default
      end
    end
  end

  config.model 'HistoryEvent' do
    label "Historial de Evento" 
    label_plural "Historial de Evento"
    list do
      field :user do
        searchable [:name, :lastname]
        label "Invitado"
          pretty_value do
            user_id = bindings[:object].user_id
            full_name = bindings[:view].full_name(user_id)
            bindings[:view].link_to "#{full_name}", bindings[:view].rails_admin.show_path('user', user_id)
          end
      end
      field :venue do
        searchable :name
        label "Sede"
            pretty_value do
            venue_id = bindings[:object].venue_id
            name = Venue.find(venue_id).name
            bindings[:view].link_to name, bindings[:view].rails_admin.show_path('venue', venue_id)
          end
      end
      field :event do
        label "Evento"
          pretty_value do
            event_id = bindings[:object].event_id
            name = Event.find(event_id).name
            bindings[:view].link_to name, bindings[:view].rails_admin.show_path('event', event_id)
          end
      end
      field :transport do
        searchable :name
        label "Transporte"
          pretty_value do
            transport_id = bindings[:object].transport_id
            if transport_id
              name = Transport.find(transport_id).name
              bindings[:view].link_to name, bindings[:view].rails_admin.show_path('transport', transport_id)
            else
              bindings[:view]
            end
          end
      end
      field :is_going do
        label "¿Asistirá?"
      end
      field :did_go do
        label "¿Asistió?"
      end
      field :created_at do
        label "Hora de Confirmación"
      end
      field :entrance_date do
        label "Hora de Entrada"
      end
    end
    show do
      field :user_id do
        label "Invitado"
      end
      field :venue_id do
        label "Sede"
      end
      field :event_id do
        label "Evento"
      end
      field :transport_id do
        label "Transporte"
      end
      field :is_going do
        label "¿Asistirá?"
      end
      field :did_go do
        label "¿Asistió?"
      end
      field :created_at do
        label "Hora de Confirmación"
      end
      field :entrance_date do
        label "Hora de Entrada"
      end
    end
    edit do
      field :user_id do
        label "Invitado"
      end
      field :venue_id do
        label "Sede"
      end
      field :event_id do
        label "Evento"
      end
      field :transport_id do
        label "Transporte"
      end
      field :is_going do
        label "¿Asistirá?"
      end
      field :did_go do
        label "¿Asistió?"
      end
      field :created_at do
        label "Hora de Confirmación"
      end
      field :entrance_date do
        label "Hora de Entrada"
      end
    end
  end

  config.model 'Role' do
    label "Rol" 
    label_plural "Roles"
    list do
      exclude_fields :id, :created_at, :updated_at
    end
    show do
      exclude_fields :id, :created_at, :updated_at
    end
  end
end

RailsAdminImport.config do |config|
  config.model User do
    excluded_fields do
      [:reset_password_token, :reset_password_sent_at, :remember_created_at,
       :sign_in_count, :current_sign_in_at, :last_sign_in_at, :current_sign_in_ip,
       :last_sign_in_ip, :version]
    end
    label :email
  end
end

**Comment by rfroetscher** [comment]
I had a similar issue. I was referring to a method in my ApplicationController that would render an unauthorized page if certain conditions were met.

In my config:
```
  config.parent_controller = 'ApplicationController'

  config.authenticate_with do
    warden.authenticate! scope: :user
    check_admin_authorization!
  end
```

In my application controller:
```
  def check_admin_authorization!
    render file: 'public/401.html', status: :unauthorized unless current_user.role == User::ADMIN_ROLE
  end
```

Basically it was trying to render some rails_admin layouts around my static HTML, which doesnt make sense.
```
  Rendering application/401.html within layouts/rails_admin/application
  Rendered application/401.html within layouts/rails_admin/application (1.3ms)
  Rendered /Users/rob/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rails_admin-1.1.1/app/views/layouts/rails_admin/_secondary_navigation.html.haml (168.7ms)
  Rendered /Users/rob/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rails_admin-1.1.1/app/views/layouts/rails_admin/_navigation.html.haml (197.0ms)
  Rendered /Users/rob/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rails_admin-1.1.1/app/views/layouts/rails_admin/_sidebar_navigation.html.haml (47.4ms)
  Rendering /Users/rob/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rails_admin-1.1.1/app/views/layouts/rails_admin/pjax.html.haml
  Rendered /Users/rob/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rails_admin-1.1.1/app/views/layouts/rails_admin/pjax.html.haml (30.8ms)
Completed 500 Internal Server Error in 956ms (ActiveRecord: 8.6ms)



ActionView::Template::Error (undefined method `breadcrumb_parent' for nil:NilClass):
     9:   .alert.alert-dismissible{class: flash_alert_class(key)}
    10:     %button.close{type: 'button', :'data-dismiss' => "alert"} &times;
    11:     = value
    12: = breadcrumb
    13: %ul.nav.nav-tabs
    14:   = menu_for((@abstract_model ? (@object.try(:persisted?) ? :member : :collection) : :root), @abstract_model, @object)
    15:   = content_for :contextual_tabs
```

Solution is to set `layout: false` on that action:
```
  def check_admin_authorization!
    render file: 'public/401.html', layout: false, status: :unauthorized unless current_user.role == User::ADMIN_ROLE
  end
```
